lib/usd/ui/layerEditor/saveLayersDialog.h

```diff
@@ -1,25 +1,43 @@
 #ifndef SAVELAYERSDIALOG_H
 #define SAVELAYERSDIALOG_H
 
+#include <pxr/usd/sdf/layer.h>
+#include <pxr/usd/usd/stage.h>
+
 #include <QtCore/QStringList>
 #include <QtWidgets/QDialog>
 #include <QtWidgets/QtWidgets>
 
-class QScrollArea;
-class QVBoxLayout;
+#include <unordered_map>
+#include <unordered_set>
+#include <vector>
+
+PXR_NAMESPACE_USING_DIRECTIVE
+
+class QWidget;
 
 namespace UsdLayerEditor {
 
-class LayerTreeItem;
+class SessionState;
 
 class SaveLayersDialog : public QDialog
 {
 public:
-    SaveLayersDialog(
-        const QString&                     title,
-        const QString&                     message,
-        const std::vector<LayerTreeItem*>& layerItems,
-        QWidget*                           in_parent);
+    typedef std::unordered_multimap<SdfLayerRefPtr, std::string, TfHash> stageLayerMap;
+
+    // Create dialog using single stage (from session state).
+    SaveLayersDialog(SessionState* in_sessionState, QWidget* in_parent);
+
+#if defined(WANT_UFE_BUILD)
+    // Create dialog for bulk save using all stages.
+    SaveLayersDialog(QWidget* in_parent, const std::vector<UsdStageRefPtr>& stages);
+#endif
```

Comment:
Previously this dialog was only used from the Layer Editor itself, but it can now be displayed from a File -> Save operation.  The first constructor is used from the LE, and the dialog gets most of the information it needs from the SessionState.  From the File -> Save callback function we have already iterated over all of the proxies and their stages and have the list of stages we need to save, so it uses this second constructor.  Saving of Usd edits does require UFE so you will see some ifdef's for that.